### DISCUSSION TITLE
qBittorrent needs an update

### DISCUSSION THREAD

**Discussion by Gangstrocity** [community-scripts/ProxmoxVE#704] [General]
Is it possible to update the qbittorrent script so that it updates the instance. Right now I'm unable to update to v5.

**Comment by tesla-76** [comment]
If it helps I report a wayto perform the update, waiting if it is possible to update the script.

1- Edit the file **/etc/apt/sources.list** by adding as the first line the string:

`deb http://ftp.debian.org/debian sid main contrib`

2- Then restart the LXC container and then proceed with the update using the commands:

`sudo apt-get update`
`sudo apt-get full-upgrade`

Confirm each update or option, when finished restart the LXC container . Obviously with each new release of qBittorrent re-execute only the last two commands to update.

**Reply by Gangstrocity** [reply]
ok, so apparently mine is ubuntu. I must have installed this one manually. Do you know what to add to the source list for ubuntu?

**Reply by tesla-76** [reply]
No, I don't know what source list to refer to for Ubuntu.

**Reply by clipz98** [reply]
> ok, so apparently mine is ubuntu. I must have installed this one manually. Do you know what to add to the source list for ubuntu?

Whether you installed the ubuntu one manually or via script, you'll need ubuntu 24.10 in order to install it since it has certain dependencies the new 5.x releases require. Trust me, been down this rabbit hole since I've been consolidating and migrating most of my LXC's to my more powerful Minisforum MS-01 the past several months. I decided to just wait for them to finally add 24.10 to the script (it's still a WIP since they're doing a bigger initiative with their [pull](https://github.com/community-scripts/ProxmoxVE/pull/643) [requests](https://github.com/community-scripts/ProxmoxVE/pull/710)). The Debian version is an unstable pre-release 5.0 version on Debian Sid, so I'd recommend not bothering with that one since the general release of 5.0.1 is out anyway. I've tried manually upgrading the LXC to 24.10 twice (once when I had 6.8 Kernel; the next attempt when I had 6.11 kernel which Ubuntu 24.10 ships with and has compatibility with). Both times my LXC died and wouldn't boot up, so I decided to keep my production version of qbit chugging along on my old server until the new script updates officially support it.

So my advice is to wait a little longer for the scripts to get updated with 24.10 as an option. If you don't want to wait, you can very easily just get the 24.10 template in Proxmox, create an LXC of it, and manually install Qbit 5.x. I'm waiting for the scripts, however, just for consistency across my homelab (which is mostly tteck scripts now on my newest server instead of only manual installs previously).

**Reply by Gangstrocity** [reply]
Thanks for the info. Yeah I tried to update my LXC and it broke everything so I restored the backup. I'll just wait it out.

**Reply by tesla-76** [reply]
@clipz98 For information. Are you sure that on Debian it is stopped at pre 5.X or did I misunderstand your statement? Because on my system it reports 5.02 (even reading the qbittorrent file update log it reports 5.02 release).

![image](https://github.com/user-attachments/assets/55459b60-2c93-4027-a07c-0596fcde5c91)

**Reply by clipz98** [reply]
> @clipz98 For information. Are you sure that on Debian it is stopped at pre 5.X or did I misunderstand your statement? Because on my system it reports 5.02 (even reading the qbittorrent file update log it reports 5.02 release).
> 
> ![image](https://private-user-images.githubusercontent.com/92620849/393397688-55459b60-2c93-4027-a07c-0596fcde5c91.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM1MTg5NzAsIm5iZiI6MTczMzUxODY3MCwicGF0aCI6Ii85MjYyMDg0OS8zOTMzOTc2ODgtNTU0NTliNjAtMmM5My00MDI3LWEwN2MtMDU5NmZjZGU1YzkxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjA2VDIwNTc1MFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTgzYzE4MmI1YzBjYjAwNWIzYTZlZWM5YjJjZDQ0NjVkOGNkZGY4NWIyZTYyZDQzODJmN2VhMjE1ZjFiNWY0MTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ALeD0KCjsgqa42vJbl-05J-CcuThuKegWIpghX7XVdE)

My info is as of over a month ago when I stopped looking at the debian versions tbh (where it was still 5.0 or something like that). Either way, I meant was the OSes 5.x had available for debian was pre-release unstable or testing versions and not available via the scripts anyway (which only support bookworm or bullseye at the moment). Those Deb versions aren't available as LXC templates either as far as I can see.

From the Deb packages source: https://packages.debian.org/search?keywords=qbittorrent&searchon=names&suite=all&section=all
![image](https://github.com/user-attachments/assets/09cce45f-4bae-47b9-a170-100487fc0db1)

**Reply by clipz98** [reply]
> Thanks for the info. Yeah I tried to update my LXC and it broke everything so I restored the backup. I'll just wait it out.

**Update**: @Gangstrocity If you're impatient like me, I decided to try out the [dev branch](https://github.com/community-scripts/ProxmoxVE/discussions/715) install of the scripts that include 24.10. It works as intended like the production scripts, but unfortunately, the this version still pulls 4.65. I guess that's what's only available from the official repos. 

To get around this, I [manually added](https://launchpad.net/~qbittorrent-team/+archive/ubuntu/qbittorrent-stable?field.series_filter=oracular) the launchpad ppa (you need to install the software properties-common first in order to add, `apt install software-properties-common`) then did an apt update and upgrade with the new PPAs. This pulled 5.0.2 and upgraded the existing install. Reboot and then congrats, you have a new 5.x instance. It defaults to beautiful dark mode. If you trust the install, move over your config and database from your old prod LXC and you should be golden!
![image](https://github.com/user-attachments/assets/413a4c6b-9dc8-4e32-bd68-f0d34318e074)

It remains to be seen how this semi-manual update will affect using the midnight commander's `update` command that you'd normally use to update the LXC apps/services, so YMMV

**Comment by kafisatz** [comment]
just cross posting this here, the following three lines work quite well on the standard qbittorrent LXC
https://github.com/community-scripts/ProxmoxVE/discussions/568#discussioncomment-11785918

**Comment by kuzyn** [comment]
My qbittorrent UI version is 5.10 and I updated qbittorrent-nox following @kafisatz's cross post. But I am still annoyed by the script not doing the update:

```  ✖️   Please create new qBittorrent LXC. Updating from v4.x to v5.x is not supported!```

What's my move? I am seeding a lot of torrent (2000+) and would like to limit migration and manual fixes, so creating a new LXC is not something I want to do.

**Reply by comporder1** [reply]
I just ran into this issue. I used claude code to trace down the issue. Here was its response: 

the ProxmoxVE community-scripts qbittorrent.sh had a leftover guard checking for a ~/.qbittorrent marker file that the install script never actually creates — so it always threw the "v4→v5 not supported" error regardless of your real version. Since your instance was already legitimately on v5, I bootstrapped the marker from the existing /opt/qBittorrent_version.txt value, which let the script's real update logic run correctly.

Here are step-by-step instructions for someone hitting the same "Please create new qBittorrent LXC. Updating from v4.x to v5.x is not supported!" error on a  Proxmox VE qBittorrent LXC installed via the community-scripts helper:

  ---
  Fix: qBittorrent update script falsely blocks update

  Cause: The community-scripts qbittorrent.sh update function checks for a marker file at ~/.qbittorrent to confirm the install is "migrated" to v5. The install  script never actually creates that file, so the check always fails and throws a scary but inaccurate "v4→v5 not supported" error — even on installs that are already running v5.

  Step 1 — Confirm you're actually affected

  Run this on the LXC/container as root:

  /opt/qbittorrent/qbittorrent-nox --version
  ls -la ~/.qbittorrent

  If the binary reports a version (e.g. qBittorrent v5.x.x) and ~/.qbittorrent does not exist, you have this exact issue.

  Step 2 — Find your current installed version

  cat /opt/qBittorrent_version.txt 2>/dev/null || /opt/qbittorrent/qbittorrent-nox --version

  Note the version number (e.g. 5.1.0).

  Step 3 — Create the missing marker file

  echo "5.1.0" > ~/.qbittorrent

  (Replace 5.1.0 with whatever version you found in Step 2. No v prefix, just the number.)

  Step 4 — Re-run the official update script

  bash -c "$(curl -fsSL https://raw.githubusercontent.com/community-scripts/ProxmoxVE/main/ct/qbittorrent.sh)"

  It will now correctly detect your current version, compare it against the latest release from userdocs/qbittorrent-nox-static on GitHub, and update if a newer one exists — stopping the service, swapping the binary, and restarting it automatically.

  Step 5 — Verify

  /opt/qbittorrent/qbittorrent-nox --version
  systemctl status qbittorrent-nox --no-pager

  Confirm the version increased and the service shows active (running).

  ---
  Note for future updates: once the marker file is created, this same update script will work correctly going forward — this is a one-time fix, not something you need to repeat each update.

**Reply by kuzyn** [reply]
Thanks for sharing @comporder1, worked here too. Effectively seems that only creating that versioned .qbittorrent file is enough to bypass the blocked update.

**Reply by goozoon** [reply]
Works :) thank you
  ✔️  Update available: qbittorrent 5.1.2 -> release-5.2.3_v2.0.14

**Comment by kuzyn** [comment]
While the message is a bit cryptic (to me at least), this seems to be intended to avoid performance issues and custom theme compatibility. See https://github.com/community-scripts/ProxmoxVE/issues/15876#issuecomment-5011953781
